pkg/oc/cli/admin/node/logs/logs.go

```diff
@@ -0,0 +1,489 @@
+		}
+		o.Selector = req.String()
+	}
+	if len(o.Selector) > 0 {
+		builder.ResourceTypes("nodes").LabelSelectorParam(o.Selector)
+	}
+	o.Builder = builder
+
+	return nil
+}
+
+func (o LogsOptions) Validate() error {
+	if len(o.Resources) == 0 && len(o.Selector) == 0 {
+		return fmt.Errorf("at least one node name or a selector (-l) must be specified")
+	}
+	if len(o.Resources) > 0 && len(o.Selector) > 0 {
+		return fmt.Errorf("node names and selector may not both be specified")
+	}
+	return nil
+}
+
+type logRequest struct {
+	node string
+	req  *rest.Request
+	err  error
+
+	raw        bool
+	skipPrefix bool
+}
+
+func (req *logRequest) WriteTo(out io.Writer) error {
+	if req.err != nil {
+		return req.err
+	}
+	err := req.writeTo(out)
+	if err != nil {
+		err = fmt.Errorf("%s %v", req.node, err)
+		req.err = err
+	}
+	return err
+}
+
+func (req *logRequest) writeTo(out io.Writer) error {
+	in, err := req.req.Stream()
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	if req.raw {
+		return optionallyDecompress(out, in)
+	}
+
+	var prefix []byte
+	if !req.skipPrefix {
+		prefix = []byte(fmt.Sprintf("%s ", req.node))
+	}
+
+	return outputDirectoryEntriesOrContent(out, in, prefix)
+}
+
+// RunLogs retrieves node logs
+func (o LogsOptions) RunLogs() error {
+	builder := o.Builder
+
+	var requests []*logRequest
+
+	var errs []error
+	result := builder.ContinueOnError().Flatten().Do()
+	err := result.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			requests = append(requests, &logRequest{node: info.Name, err: err})
+			return nil
+		}
+		mapping := info.ResourceMapping()
+		client, err := o.RESTClientGetter(mapping)
+		if err != nil {
+			requests = append(requests, &logRequest{node: info.Name, err: err})
+			return nil
+		}
+		path := client.Get().
+			Namespace(info.Namespace).Name(info.Name).
+			Resource(mapping.Resource.Resource).SubResource("proxy", "logs").Suffix(o.Path).URL().Path
+		if strings.HasSuffix(o.Path, "/") {
+			path += "/"
+		}
+
+		req := client.Get().RequestURI(path).
+			SetHeader("Accept", "text/plain, */*").
+			SetHeader("Accept-Encoding", "gzip")
+		if o.Path == "journal" {
+			if len(o.UntilTime) > 0 {
+				req.Param("until", o.UntilTime)
+			}
+			if len(o.SinceTime) > 0 {
+				req.Param("since", o.SinceTime)
+			}
+			if len(o.Output) > 0 {
+				req.Param("output", o.Output)
+			}
+			if len(o.Units) > 0 {
+				for _, unit := range o.Units {
+					req.Param("unit", unit)
+				}
+			}
+			if len(o.Pattern) > 0 {
+				req.Param("grep", o.Pattern)
+				req.Param("case-sensitive", fmt.Sprintf("%t", !o.PatternCaseInsensitive))
+			}
+			if o.Tail > 0 {
+				req.Param("tail", strconv.Itoa(o.Tail))
+			}
+		}
+
+		requests = append(requests, &logRequest{
+			node: info.Name,
+			req:  req,
+			raw:  o.Raw || o.Path == "journal",
+		})
+		return nil
+	})
+	if err != nil {
+		if agg, ok := err.(errors.Aggregate); ok {
+			for _, err := range agg.Errors() {
```

Comment:
Why not just `append(errs, agg.Errors()...)` ?

Reply:
Yeah